h2o-algos/src/main/java/hex/ensemble/StackedEnsemble.java

```diff
@@ -78,7 +78,9 @@ public void init(boolean expensive) {
       throw new H2OIllegalArgumentException("Setting \"distribution\" to StackedEnsemble is unsupported. Please set it in \"metalearner_parameters\".");
     }
 
-    checkFoldColumnPresent(_parms._metalearner_fold_column, train(), valid(), _parms.blending());
+    checkColumnPresent("fold", _parms._metalearner_fold_column, train(), valid(), _parms.blending());
```

Comment:
is it not a good occasion to deprecate those `metalearner_fold_column`, `metalearner_nfolds`, `metalearner_fold_assignment` parameters here if we start accepting the param values that are set directly on the SE model?

It would really simplify API and avoid some confusions:
For example, with your current changes, if user does the following
```
H2OStackedEnsembleEstimator(fold_column="fcol", weights_column="wcol")
```
then it will use the weights correctly but ignore the fold column (actually raise an error in Python, but when used directly in Java, it will just ignore it).
so that user needs to do:
```
H2OStackedEnsembleEstimator(metalearner_fold_column="fcol", weights_column="wcol")
```
which looks really inconsistent.
I never really understood this `_metalearner_xxx` naming for those CV specific arguments and we should go towards SE being more consistent and behaving more like other algos.

Note: we have utilities to deprecate properties in Py. For arguments (Py and R) we could emit a standard warning when they're passed.

Idea of deprecation here also involves creation of ticket to remove completely support for those 3 properties in let's say version 3.36.
@ledell , @tomasfryda any thoughts?

Reply:
I agree that we should keep things consistent with the rest of the algos, that includes deprecating `metalearning_fold_column` etc.

Reply:
@ledell I'd say that the status of this PR is dependent on the decision if we will want to deprecate `metalearner_*` params in this PR.

Reply:
it doesn't have to be in this PR, but it would be great if both are done in the same release because all those parameters (sometimes with, sometimes without prefix) are getting confusing.
Deprecations should be done only for `3.32.0.1`, so do we really need this in `zahradnik`?

Reply:
I'd suggest to consider merging this to `master` and add a follow-up ticket for the `metalearner` fold params deprecations and ensure it is done before `3.32`.

Reply:
I agree. I created a JIRA[1] for deprecation of the `metalearner_*` params. So if @ledell is ok with this being in `master` instead of `rel-zahradnik`, I will merge it.


[1] https://0xdata.atlassian.net/browse/PUBDEV-7675